### ISSUE TITLE
Add a relationship from SCOs to Infrastructure

### ISSUE THREAD

**Issue by ejratl** [oasis-tcs/cti-stix2#151]
Use case: you have the IPv4 address and you want to know what Infrastructure it is part of.

Given a SCO, such as an IPv4 address, the desire is to have a relationship that directly links the SCO to the associated Infrastructure object, in addition to the relationship from the Infrastructure object to the SCO. 

Adding this relationship adds bi-directional relationships between Infrastructure and SCOs.

**Comment by ikiril01** [comment]
I imagine we'd want something similar for SCO -> Malware, so could we make this a generic relationship that works for any such case? Maybe something like SCO -> `associated_with` -> SDO.

**Comment by jordan2175** [comment]
This is really about doing bi-directional relationships, something we have avoided all over in STIX.  We have tried to use a common voice / verb tense on relationships and define them from the object that you would most likely start with.  But relationships by themselves just connect two things together and have a friendly name to help give it some context.  We will talk about this on an upcoming working call.
